test/e2e/aws_test.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"flag"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"github.com/onsi/gomega/types"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awssts "github.com/aws/aws-sdk-go/service/sts"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	capa "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1alpha1"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/awserrors"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/cloudformation"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/sts"
+	"sigs.k8s.io/cluster-api-provider-aws/test/e2e/util/kind"
+	capi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+const (
+	kindTimeout      = 5 * 60
+	namespace        = "capa-test"
+	clusterName      = "capa-test-cluster"
+	controlPlaneName = "capa-test-control-plane"
+
+	kubeletVersion = "v1.13.2"
```

Comment:
Would it be possible to pull kubeletVersion from the example manifests? Currently we only really publish one supported version and I worry that this might get out of sync and be confusing to troubleshoot over time.

Reply:
Yes, but it'll take a bit of work. Make this a TODO and issue and can see how to handle it later. Also applies to the image generation

Reply:
TOO LATE did it anyway

Reply:
@randomvariable what do you mean by image generation? do you mean the AMIs? We're already using compile-time docker images from Bazel.

Reply:
AMIs. Related to https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/576#pullrequestreview-203817286